torch/jit/__init__.py

```diff
@@ -1526,21 +1541,28 @@ def __init__(self, original, stubs):
             # __init__ can run correctly
             self.__dict__['_initialized'] = False
             super(WeakScriptModuleProxy, self).__init__()
-
             # Store a weak reference to the original module
             self.__dict__["_original"] = weakref.ref(original)
 
             constants_set = set(getattr(original, "__constants__", []))
             self.__dict__["_constants_set"] = {}
 
+            if not hasattr(original, '_parameters'):
+                raise RuntimeError("'{}' has not been initialized, did you forget to call 'super()'?"
+                                   .format(type(original).__name__))
+
             # Copy Parameters and Modules
             for name in dir(original):
                 item = getattr(original, name)
                 if item is None and name in original._parameters:
                     # XXX: treat None value simply as module attributes instead of adding them to the parameter list
                     # TODO: need to handle this more generally when non-tensor attributes added to module
                     object.__setattr__(self, name, item)
-                elif isinstance(item, Parameter) or (isinstance(item, Module) and item is not self):
+                elif item is self:
```

Comment:
I can't follow this code here. Not sure why it is deciding to ignore some things, and not others.